core/src/test/java/com/graphhopper/coll/MapEntry_DiffblueTest.java

```diff
@@ -0,0 +1,143 @@
+/**
+ *
+ *
+ * Copyright 2018 Diffblue Limited
+ *
+ * Diffblue Limited licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.graphhopper.coll;
+
+import com.diffblue.deeptestutils.Reflector;
+import com.graphhopper.coll.MapEntry;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.lang.reflect.InvocationTargetException;
+
+public class MapEntry_DiffblueTest {
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  /* testedClasses: MapEntry */
+
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers:
+   *  - conditional line 63 branch to line 66
+   *  - conditional line 66 branch to line 67
+   */
+
+  @Test
+  public void equalsInputNotNullOutputFalse2() throws InvocationTargetException {
+
+    // Arrange
+    final MapEntry objectUnderTest = new MapEntry(null, null);
+    final MapEntry obj = (MapEntry)Reflector.getInstance("com.graphhopper.coll.MapEntry");
+    Reflector.setField(obj, "value", 0);
+    Reflector.setField(obj, "key", null);
+
+    // Act
+    final boolean retval = objectUnderTest.equals(obj);
```

Comment:
@peterschrammel Can we work on naming `objectUnderTest` and `obj` something like `mapEntry` and `mapEntry2`?

Reply:
We had that before and it was less readable because it is harder to identify the object under test if it is not called that way.